### PR TITLE
Fix dune subdir stanza syntax highlighting

### PR DESCRIPTION
**Pull request by SaySayo**
Fixes (#864)

### CODE REVIEW

**Comment by tmattio** [comment]
Thanks @SaySayo 🙂 

The `subdir` stanza works by evaluating the dune stanza in a subdirectory. So it should support *every* dune stanza, not only the rule one. To do this I think we can extract all the patterns in a scope, and include this in the `subdir` patterns (assuming the textmate grammar support recursivity).

**Comment by SaySayo** [comment]
That's a great idea! I'll do that now

File: syntaxes/dune.json [hunk 1]
```diff
@@ -1235,9 +1235,214 @@
...
+                      "include": "#general"
+                    }
+                  ]
+                },
                 {
                   "include": "#action"
                 },
+                {
+                  "include": "#variables"
+                },
```
Comment (mnxn) [inline syntaxes/dune.json:1445]: What's the point of including `#variables` here?

The rules in "variables" are just for highlighting the name in between `%{` and `}`.
The actual syntax for the full variable `%{...}` is already included in `#general`.

**Comment by mnxn** [comment]
> The `subdir` stanza works by evaluating the dune stanza in a subdirectory. So it should support _every_ dune stanza, not only the rule one. To do this I think we can extract all the patterns in a scope, and include this in the `subdir` patterns (assuming the textmate grammar support recursivity).

I think extracting all of the patterns into a new scope is actually unnecessary. You can make the patterns of `subdir` include `$self` and that should highlight everything

I believe this should work:
```json
"comment": "subdir",
"begin": "\\([[:space:]]*(subdir)\\b",
"end": "\\)",
"beginCaptures": {
  "1": {
    "name": "keyword.language.dune"
  }
},
"patterns": [{ "include": "$self" }]
```

File: syntaxes/dune.json [hunk 2]
```diff
@@ -1225,6 +1222,9 @@
             }
           },
           "patterns": [
+            { 
+              "include": "$self" 
+            },
             {
```
Comment (mnxn) [inline syntaxes/dune.json:1228]: I think the rest of this is unnecessary now since "$self" already recognizes a `rule`.

Comment (SaySayo) [reply syntaxes/dune.json:1228]: Oh right, Thank you

**Review by mnxn — APPROVED** [review]

**Comment by mnxn** [comment]
Thanks!

**Pull request merged**